### PR TITLE
refactor collateral auction

### PR DESCRIPTION
**Pull request by wangjj9219**

### CODE REVIEW

**Comment by wangjj9219** [comment]
resolve #320

File: modules/cdp_engine/src/lib.rs [hunk 1]
```diff
@@ -713,9 +713,8 @@ impl<T: Trait> RiskManager<T::AccountId, CurrencyId, Balance, Balance> for Modul
 		collateral_balance: Balance,
 		debit_balance: Balance,
 	) -> DispatchResult {
-		let debit_value = Self::get_debit_value(currency_id, debit_balance);
-
-		if !debit_value.is_zero() {
+		if !debit_balance.is_zero() {
```
Comment (wangjj9219) [inline modules/cdp_engine/src/lib.rs:716]: fix bug maybe lead tiny debit balance bypass MininumDebitBalance check due to caculation loss

File: modules/auction_manager/src/lib.rs [hunk 2]
```diff
@@ -59,7 +59,9 @@ pub struct CollateralAuctionItem<AccountId, BlockNumber> {
 	/// current collateral amount for sale
 	#[codec(compact)]
 	amount: Balance,
-	/// Target sales amount want to get by this auction
+	/// Target sales amount want to get by this auction,
+	/// if zero, collateral auction will never be reverse stage,
+	/// otherwise, target amount is the actual payment amount of active bidder
```
Comment (wangjj9219) [inline modules/auction_manager/src/lib.rs:64]: Consider the case which the target is zero, although it is unlikely to happen in the system. If target is zero, the collateral auction will never in reverse stage.

File: modules/cdp_treasury/src/lib.rs [hunk 3]
```diff
@@ -415,30 +415,39 @@ impl<T: Trait> CDPTreasuryExtended<T::AccountId> for Module<T> {
...
+			let lots_count = if max_auctions_count.is_zero()
+				|| collateral_auction_maximum_size.is_zero()
+				|| amount <= collateral_auction_maximum_size
+			{
+				One::one()
+			} else {
+				let mut count = amount
+					.checked_div(collateral_auction_maximum_size)
+					.expect("collateral auction maximum size is not zero; qed");
+				if !(amount % collateral_auction_maximum_size).is_zero() {
```
Comment (xlc) [inline modules/cdp_treasury/src/lib.rs:435]: use `checked_rem`

**Pull request merged**